src/gwt/src/org/rstudio/core/client/jsonrpc/RpcResponse.java

```diff
@@ -37,19 +60,27 @@ public final static RpcResponse parse(String json)
       }
       catch(Exception e)
       {
-         try
+         if (strict)
          {
-            // there are some cases where json emitted by our 
-            // server isn't parsable by parseStrict (for example,
-            // see bug #3025). for these situations we call 
-            // parseLenient (which in turn calls eval)
-            @SuppressWarnings("deprecation")
-            JSONValue val = JSONParser.parseLenient(json);
-            return val.isObject().getJavaScriptObject().cast();
+            // in strict mode, don't try eval
+            return null;
          }
-         catch(Exception e2)
+         else
          {
-            return null;
+            try
+            {
+               // there are some cases where json emitted by our
+               // server isn't parsable by parseStrict (for example,
+               // see bug #3025). for these situations we call
```

Comment:
Nit: I know you didn't add this, but I don't know what bug database this references! Comment is 9 years old so FogBugz? Maybe remove this comment (or point it at something meaningful, if such a thing can be found?).

Reply:
Yup, must be FogBugz. I'll just remove it.